### PR TITLE
[Kernel] Flashinfer correctness fix for v0.1.3

### PR DESCRIPTION
**Pull request by LiuXiaoxuanPKU**
Reported by @felixzhu555 , `is_profile_run` is buggy in flashinfer backend, which will fail flashinfer v0.1.3. This PR fixes this, and update CI flashinfer version to v1.0.3.

### CODE REVIEW

File: .buildkite/test-pipeline.yaml [hunk 1]
```diff
@@ -61,7 +61,7 @@ steps:
   - tests/basic_correctness
   commands:
   # This flashinfer installation will fail on AMD ROCm, so it is set as optional.
-  - pip install https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.2/flashinfer-0.1.2+cu121torch2.4-cp310-cp310-linux_x86_64.whl || true
+  - pip install https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.3/flashinfer-0.1.3+cu124torch2.4-cp310-cp310-linux_x86_64.whl || true
```
Comment (comaniac) [inline .buildkite/test-pipeline.yaml:64]: Just update the Dockerfile and remove all of these.

File: vllm/attention/backends/flashinfer.py [hunk 2]
```diff
@@ -141,23 +142,18 @@ def begin_forward(self):
...
             batch_size = self.query_start_loc.shape[0] - 1
             assert batch_size >= 0
-            # The profile run does not read kv cache.
-            # Both paged_kv_indices and paged_kv_last_page_len are empty.
-            # paged_kv_indptr is a zero tensor with size batch_size + 1.
-            if self.is_profile_run:
-                self.paged_kv_indptr = torch.zeros(batch_size + 1,
-                                                   device=self.device)
-            else:
+            # Only use flashinfer in the non-profile run
```
Comment (comaniac) [inline vllm/attention/backends/flashinfer.py:145]: This comment is a bit confusing as we are in the FlashInfer backend. Can we say more about why we skip the following logic in profile run and what's the outcome?

**Review by comaniac — APPROVED** [review]
LGTM. Just a nit

**Comment by comaniac** [comment]
CI failure seems like a real bug

```
[2024-08-09T04:52:05Z]   File "/usr/local/lib/python3.10/dist-packages/flashinfer/prefill.py", line 791, in begin_forward
--
  | [2024-08-09T04:52:05Z]     self._wrapper.begin_forward(
  | [2024-08-09T04:52:05Z] RuntimeError: CHECK_EQ(paged_kv_indptr.size(0), batch_size + 1) failed. 1 vs 257
```

**Comment by exceedzhang** [comment]
@LiuXiaoxuanPKU It's bugs
image

**Pull request merged**